# Mapping health needs
Data from GBD 2010 study

In [1]:
setwd('/media/igna/Elements/HotelDieu/Cochrane')

GBD <- read.table("Mapping_Cancer/Tables/GBD_data_per_country_and_28_diseases_2005.txt")
names(GBD)


[1] "causelevel1"  "causelevel2"  "causelevel3"  "causelevel4"  "causelevel5" 
 [6] "country_name" "region"       "year"         "nm_mean"      "rt_mean"     
[11] "Disease"      "Sup_region"

In [2]:
dis <- levels(GBD$Dis)
reg <- levels(GBD$Sup_reg)

S <- tapply(GBD$nm_mean,paste(GBD$Sup_region,GBD$Disease,sep="&"),sum)
L <- strsplit(names(S),"&")

M <- matrix(NA,ncol=length(reg),nrow=length(dis))

for(i in 1:length(S)){
M[dis==L[[i]][2],reg==L[[i]][1]] <- S[i]
}

M <- data.frame(M)
names(M) <- reg
rownames(M) <- dis

M$Tot <- apply(M,1,function(x){sum(x[!is.na(x)])})

write.table(M,"Mapping_Cancer/Tables/GBD_data_per_region_and_28_diseases_2005.txt")

In [3]:
head(M)

Central Europe, Eastern Europe, and Central Asia
Cardiovascular and circulatory diseases                                                                                          53161407
Chronic respiratory diseases                                                                                                      5644608
Cirrhosis of the liver                                                                                                            4090337
Congenital anomalies                                                                                                              2079423
Diabetes, urinary diseases and male infertility                                                                                   5049825
Diarrhea, lower respiratory infections, meningitis, and other common infectious diseases                                          7281616
                                                                                         High-income
Cardiovascular and circulatory diseases                                                     43545779
Chronic respiratory diseases                                                                13641980
Cirrhosis of the liver                                                                       3995636
Congenital anomalies                                                                         2296894
Diabetes, urinary diseases and male infertility                                             12338425
Diarrhea, lower respiratory infections, meningitis, and other common infectious diseases     6847691
                                                                                         Latin America and Caribbean
Cardiovascular and circulatory diseases                                                                     15012902
Chronic respiratory diseases                                                                                 5500905
Cirrhosis of the liver                                                                                       2376422
Congenital anomalies                                                                                         3617508
Diabetes, urinary diseases and male infertility                                                              6866404
Diarrhea, lower respiratory infections, meningitis, and other common infectious diseases                    10635120
                                                                                         North Africa and Middle East
Cardiovascular and circulatory diseases                                                                      19214033
Chronic respiratory diseases                                                                                  4781865
Cirrhosis of the liver                                                                                        1471645
Congenital anomalies                                                                                          5091068
Diabetes, urinary diseases and male infertility                                                               4671200
Diarrhea, lower respiratory infections, meningitis, and other common infectious diseases                     10785476
                                                                                         South Asia
Cardiovascular and circulatory diseases                                                    56395260
Chronic respiratory diseases                                                               38910999
Cirrhosis of the liver                                                                      7748645
Congenital anomalies                                                                       12842242
Diabetes, urinary diseases and male infertility                                            18349709
Diarrhea, lower respiratory infections, meningitis, and other common infectious diseases  130854936
                                                                       

In [4]:
#In the 28 groups of diseases, were already supressed Other infectious diseases and Other endocrine...
dim(M)
#Database with the total burden (all disease) per region
#We verify the share of the two residual categories among total burden and among diseases

GBT <- read.table("Mapping_Cancer/Tables/Nb_DALYs_per_region_and_disease_all_periods.txt")
names(GBT)
head(GBT)

[1] 28  8

[1] "Region"     "Tot"        "Chronic"    "Neoplasms"  "Infectious"
[6] "MNN"        "Oth_cat"    "No_GBD"     "year"

Region      Tot   Chronic Neoplasms
5                                       South Asia 697.8030 179.78026 22.858406
7                              Sub-Saharian Africa 596.5828  75.73523  9.557038
6            Southeast Asia, East Asia and Oceania 529.4901 196.38049 65.739058
2                                      High-income 259.4157 139.11926 43.298284
1 Central Europe, Eastern Europe, and Central Asia 177.1321  83.15489 19.507981
3                      Latin America and Caribbean 135.9614  54.56716  9.697365
  Infectious        MNN   Oth_cat    No_GBD year
5 189.877884 141.169630 155.38083 8.7360229 2005
7 322.082215 105.964212  74.66367 8.5804634 2005
6  69.921409  39.639594 153.72809 4.0814231 2005
2   9.133358   5.137104  60.68483 2.0428897 2005
1  13.187724   6.157954  54.55410 0.5694141 2005
3  17.633272  14.885091  37.91905 1.2594164 2005

In [5]:
1-sum(M$Tot)/(sum(GBT$Tot[GBT$year==2005])*1e6)

[1] 0.01059768

Ok, less than 1.1% of the global total burden in 2005

In [6]:
#Share of Other categories among diseases (no Injuries) and across regions
Oth_cat_gbd <- sum(GBT$Tot[GBT$year==2005])*1e6 - sum(M$Tot)
GBD_diseases <- sum(GBT$Tot[GBT$year==2005])*1e6 - sum(M$Tot[rownames(M)=="Injuries"])
Oth_cat_gbd/GBD_diseases

[1] 0.01185066

Ok, less than 1.2% of the burden of diseases (no Injuries)

In [7]:
#Across regions
GBT <- GBT[order(GBT$Region),]
Oth_cat_gbd <- GBT$Tot[GBT$year==2005]*1e6 - apply(M[-ncol(M)],2,sum)
GBD_diseases <- GBT$Tot[GBT$year==2005]*1e6 - M[rownames(M)=="Injuries",-ncol(M)]
100*Oth_cat_gbd/GBD_diseases

Central Europe, Eastern Europe, and Central Asia High-income
Injuries                                        0.3743995   0.8745466
         Latin America and Caribbean North Africa and Middle East South Asia
Injuries                    1.078574                     1.307661   1.410292
         Southeast Asia, East Asia and Oceania Sub-Saharian Africa
Injuries                             0.8776721            1.542873

Residual categories have maximised GBD in SSA <= 1.6% of total burden of diseases

In [8]:
#What is the part of injuries among the total burden
sum(M$Tot[rownames(M)=="Injuries"])/(sum(GBT$Tot[GBT$year==2005])*1e6)

[1] 0.1057308

Injuries = 10.6% of total burden

In [9]:
#Burden of injuries across regions
M[rownames(M)=="Injuries",-ncol(M)]/(GBT$Tot[GBT$year==2005]*1e6)

Central Europe, Eastern Europe, and Central Asia High-income
Injuries                                        0.1413904  0.09953742
         Latin America and Caribbean North Africa and Middle East South Asia
Injuries                   0.1411765                    0.1061304  0.1122882
         Southeast Asia, East Asia and Oceania Sub-Saharian Africa
Injuries                             0.1217432           0.0677987

Injuries maximum in CEE and LATAM = 14.1% Minimum in SubSaharian Africa = 6.8%

In [10]:
#We supress Injuries from the burden
M <- M[rownames(M)!="Injuries",]
dim(M)

[1] 27  8

In [13]:
#Burden per region
apply(M,2,sum)/1e6
apply(M,2,sum)/sum(M$Tot)

Central Europe, Eastern Europe, and Central Asia 
                                        151.5179 
                                     High-income 
                                        231.5513 
                     Latin America and Caribbean 
                                        115.5074 
                    North Africa and Middle East 
                                        102.2734 
                                      South Asia 
                                        610.7119 
           Southeast Asia, East Asia and Oceania 
                                        460.9468 
                             Sub-Saharian Africa 
                                        547.5548 
                                             Tot 
                                       2220.0635 
                                         Prop_bd 
                                          0.0001

Central Europe, Eastern Europe, and Central Asia 
                                    6.824934e-02 
                                     High-income 
                                    1.042994e-01 
                     Latin America and Caribbean 
                                    5.202887e-02 
                    North Africa and Middle East 
                                    4.606778e-02 
                                      South Asia 
                                    2.750876e-01 
           Southeast Asia, East Asia and Oceania 
                                    2.076278e-01 
                             Sub-Saharian Africa 
                                    2.466393e-01 
                                             Tot 
                                    1.000000e+00 
                                         Prop_bd 
                                    4.504376e-08

In [12]:
#Burden per disease
M$Prop_bd <- 100*M$Tot/(sum(M$Tot))
head(M[order(M$Tot,decreasing=TRUE),c(ncol(M)-1:0)])

Tot
Diarrhea, lower respiratory infections, meningitis, and other common infectious diseases 329285537
Cardiovascular and circulatory diseases                                                  287404109
Neonatal disorders                                                                       220288076
Neoplasms                                                                                176868544
Mental and behavioral disorders                                                          171914381
Musculoskeletal disorders                                                                151554779
                                                                                           Prop_bd
Diarrhea, lower respiratory infections, meningitis, and other common infectious diseases 14.832258
Cardiovascular and circulatory diseases                                                  12.945761
Neonatal disorders                                                                        9.922602
Neoplasms                                                                                 7.966824
Mental and behavioral disorders                                                           7.743670
Musculoskeletal disorders                                                                 6.826597

In [15]:
tail(M[order(M$Tot,decreasing=TRUE),c(ncol(M)-1:0)])

Tot      Prop_bd
Oral disorders                              14336517.776 0.6457706145
Sexually transmitted diseases excluding HIV 12922191.897 0.5820640641
Hepatitis                                   12551046.126 0.5653462644
Gynecological diseases                       9710285.172 0.4373877200
Sudden infant death syndrome                 1877956.449 0.0845902128
Leprosy                                         8539.553 0.0003846535

In [19]:
8.539553e+03/1e6

[1] 0.008539553

In [20]:
write.table(M,"Mapping_Cancer/Tables/GBD_data_per_region_and_27_diseases_2005.txt")